src/components/cell/Cell.jsx

```diff
@@ -0,0 +1,120 @@
+import './Cell.css';
+
+import { constants } from '../../app/config';
+import { useStateValue } from '../../app/state/useStateValue';
+import { arrayIsInArray, generateRandomRgbColor } from '../../utils/main';
+import player1 from '../../assets/player-1.svg';
+import bottle2 from '../../assets/bottle-2.svg';
+
+export default function Cell ({ position }) {
+  const [x, y] = position;
+  const [state] = useStateValue();
+
+  const drawThing = (x, y, player, bottle, showPos = false) => {
```

Comment:
nit: it is better to render the player and bottle directly, instead of checking each cell for the presence

And the same for gpgp and protected zone

Reply:
this will need a big refactor in whole render cells approach
it is a requirement or a suggestion?

Reply:
just a suggestion :)

Reply:
great

I hope you don't mind but I'll leave it as it is, 
this change would radically change the way I implemented the feature, 
and that would force me to redo all the code and delay the challenge even more

thanks for understanding @realiarthur